### PR TITLE
Use float in avar calculation instead ints and checking their overflows

### PR DESCRIPTION
**Pull request by ebraminio**
<s>The problem is in `test-var-coords.c`, the conversion error with it should be turned from 0.5 > to 0.8 > to allow tests be passed.</s> fixed by the comment below

### CODE REVIEW

File: src/hb-ot-var-avar-table.hh [hunk 1]
```diff
@@ -89,14 +89,9 @@ struct SegmentMaps : ArrayOf<AxisValueMap>
...
-    int factor;
-    if (hb_signed_mul_overflows (arrayZ[i].toCoord - arrayZ[i-1].toCoord,
-				 value - arrayZ[i-1].fromCoord,
-				 factor))
-      return arrayZ[i-1].toCoord;
-
     int denom = arrayZ[i].fromCoord - arrayZ[i-1].fromCoord;
-    return arrayZ[i-1].toCoord + (factor + denom/2) / denom;
+    return roundf (arrayZ[i-1].toCoord + ((float) (arrayZ[i].toCoord - arrayZ[i-1].toCoord) *
+					  (value - arrayZ[i-1].fromCoord) + denom/2) / denom);
```
Comment (behdad) [inline src/hb-ot-var-avar-table.hh:94]: You should remove the `+ denom/2` now that you are calling `roundf`.

Comment (ebraminio) [reply src/hb-ot-var-avar-table.hh]: Thanks! Apparently applying that fixes its accuracy issue.

Comment (ebraminio) [reply src/hb-ot-var-avar-table.hh]: So now to decide which way to go, this or #2322

**Comment by behdad** [comment]
I'm happy with the float version.  As I said, since there's a division involved, float is probably faster even.

**Comment by ebraminio** [comment]
Thanks!

**Pull request merged**